src/elf/dyn.rs

```diff
@@ -10,6 +10,9 @@ macro_rules! elf_dyn {
             /// Integer value
             pub d_val: $size,
         }
+
+        use plain;
+        unsafe impl plain::Plain for Dyn {}
```

Comment:
You can checkout https://github.com/m4b/scroll_derive for a very simple derive implementation, but:

1. I don't know if custom derive works with `core` (probably does, don't see why not)?
2. both `syn` and `quote` will add like 5-10 seconds to compile time

I don't allow the `Pread`, `Pwrite` derives on the `no_std` portion of goblin for this reason, even though they would be useful, as the compile time is almost instantaneous now with a `elf32`, `elf64` feature flags, etc., in addition to adding 2 upstream dependencies that seem to keep profilerating (e.g., recent `syn` now adds some crate called `synom` as a dependency, etc.)